explorer/app2/js/app.js

```diff
@@ -13,253 +13,382 @@ const competitors = [
+            $('.search-button').each(function() {
+              $(this).attr('disabled', true);
+            });
+          });
+    });
+  })
+  $('.search-button').each(function() {
+    $(this).on('click', function() {
+      getPageOfTransactions($(this).prev().val(), 1);
+    });
   });
+}
 
-function setupCharts() {
-  charts.push(
-    initializeChart('canvas-1'),
-    initializeChart('canvas-2'),
-    initializeChart('canvas-3')
-  );
+function showPage() {
+  var newState = currentMetricsResult.state;
+  if (newState !== currentState) {
+    switch (newState) {
+      case STATE_UNKNOWN:
+        forceUpdateTickers = true;
+        currentState = newState;
+        $('.page-1').hide();
+        $('.page-2').hide();
+        $('.page-3').hide();
+        $('.page-0').show();
+        break;
+      case STATE_STARTED:
+        forceUpdateTickers = true;
+        currentState = newState;
+        $('.page-0').hide();
+        $('.page-2').hide();
+        $('.page-3').hide();
+        $('.page-1').show();
+        break;
+      case STATE_FINISHED:
+        forceUpdateTickers = true;
+        currentState = newState;
+        $('#top-buttons').show();
+        $('.page-0').hide();
+        $('.page-1').hide();
+        $('.page-3').hide();
+        $('.page-2').show();
+        break;
+      case STATE_TERMINATED:
+        forceUpdateTickers = true;
+        currentState = newState;
+        $('#top-buttons').show();
+        $('.page-0').hide();
+        $('.page-1').hide();
+        $('.page-2').hide();
+        $('.page-3').show();
+        break;
+    }
+  }
 }
 
-function initializeChart(id) {
-  return $('#' + id).chart({
-      width: 512,
-      height: 512,
-      radius: 228,
-      textSize: '80px',
-      textColor: '#5B76F1',
-      innerlineHeight: 8,
-      outerLineHeight: 54,
-      glowLineHeight: 36,
-      parent: $('#' + id).parent(),
-  }, 10, 1500);
+function updatePages() {
+  showPage();
+  updateTickers();
+  updateStats();
+  updateGraphs();
+  updateCompetitors();
 }
 
-function updateTickers(classSuffix, future) {
-  const tickerClass = '.counter-' + classSuffix;
-  if (tickerClass === currentTickerClass) {
-    return;
+function updateTickers() {
+  if (forceUpdateTickers) {
+    forceUpdateTickers = false;
+    const future = getTickerOffset();
+    const now = new Date().getTime();
+    const diffSeconds = future > now ? Math.round((future - now) / 1000) : 0;
+    $('.ticker').each(function() {
+      $(this).FlipClock(diffSeconds, {
+        clockFace: 'HourlyCounter',
+        countdown: true
+      });
+    });
   }
+}
 
-  const now = new Date().getTime();
-  const diff = future > now ? Math.round((future - now) / 1000) : 0; // seconds
-
-  $(tickerClass).each(function() {
-    $(this).FlipClock(diff, {
-      clockFace: 'HourlyCounter',
-      countdown: true
-    });
+function updateStats() {
+  $('.tps-spot').each(function() {
+    $(this).text(beautifulNumber(currentMetricsResult.spot));
+  });
```

Comment:
I might be wrong, but I think all of these can be rewritten as 
`$('.tps-spot').text(beautifulNumber(currentMetricsResult.spot));`

Reply:
Yes, I was hoping for that "implied iteration" to just be there, like, always. But sometimes (especially with the countdown tickers) this trick just doesn't work :( I'll look into this if/as I get a couple of hours to spare, though (we're getting dangerously close to Java levels of verbosity...)